src/features/metrics/aggregate/index.js

```diff
@@ -28,13 +38,80 @@ export class Aggregate extends AggregateBase {
     drain(this.agentIdentifier, this.featureName)
   }
 
-  storeMetric (type, name, params, value) {
+  storeSupportabilityMetrics (name, value) {
     if (this.blocked) return
+    const type = SUPPORTABILITY_METRIC
+    const params = { name }
     this.aggregator.storeMetric(type, name, params, value)
   }
 
-  storeEventMetrics (type, name, params, metrics) {
+  storeEventMetrics (name, metrics) {
     if (this.blocked) return
+    const type = CUSTOM_METRIC
+    const params = { name }
     this.aggregator.store(type, name, params, metrics)
   }
+
+  singleChecks () {
+    // report generic info about the agent itself
+    // note the browser agent version
+    this.storeSupportabilityMetrics(`Generic/Version/${VERSION}/Detected`)
+    // report loaderType
+    const { loaderType } = getRuntime(this.agentIdentifier)
+    if (loaderType) this.storeSupportabilityMetrics(`Generic/LoaderType/${loaderType}/Detected`)
+
+    // frameworks on page
+    if (isBrowserScope) {
+      onDOMContentLoaded(() => {
+        getFrameworks().forEach(framework => {
+          this.storeSupportabilityMetrics('Framework/' + framework + '/Detected')
+        })
+      })
+    }
+
+    // file protocol detection
+    if (protocol.isFileProtocol()) {
+      this.storeSupportabilityMetrics('Generic/FileProtocol/Detected')
+      protocol.supportabilityMetricSent = true
+    }
+
+    // obfuscation rules detection
+    const rules = getRules(this.agentIdentifier)
+    if (rules.length > 0) this.storeSupportabilityMetrics('Generic/Obfuscate/Detected')
+    if (rules.length > 0 && !validateRules(rules)) this.storeSupportabilityMetrics('Generic/Obfuscate/Invalid')
+  }
+
+  eachSessionChecks () {
+    if (!isBrowserScope) return
+
+    // [Temporary] Report restores from BFCache to NR1 while feature flag is in place in lieu of sending pageshow events.
+    windowAddEventListener('pageshow', (evt) => {
+      if (evt.persisted) { this.storeSupportabilityMetrics('Generic/BFCache/PageRestored') }
+      return
+    })
+  }
+
+  unload () {
+    // Page Resources detection for estimations with resources feature work
+    // TODO - these SMs are to be removed when we implement the actual resources feature
+    try {
+      if (this.resourcesSent) return
+      // make sure this only gets sent once
+      this.resourcesSent = true
+      // differentiate between internal+external and ajax+non-ajax
+      const ajaxResources = ['beacon', 'fetch', 'xmlhttprequest']
+      const internalUrls = ['nr-data.net', 'newrelic.com', 'nr-local.net', 'localhost']
+      function isInternal (x) { return internalUrls.some(y => x.name.indexOf(y) >= 0) }
```

Comment:
Minor, but you _could_ use the existing Array.prototype.find polyfill rather than add the .some, but yeah, you can ignore this.

Reply:
We will likely need to find some balance between excluding our syntax and adding to the polyfill...  We should investigate more auto-propagation of the polyfill file at a later time so that we aren't bound by these decisions.